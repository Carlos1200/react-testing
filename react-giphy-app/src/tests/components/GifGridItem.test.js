import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>', () => { 
    const title = 'Titulo del GIF';
    const url = 'https://media.giphy.com/media/CsdYnIX8x44pdUeLBd/giphy-downsized-large.gif';
    
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('Se debe mostrar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
     })

    test('debe de tener un parrafo con el title', () => { 
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener una imagen con la url y alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn');
    })
 })