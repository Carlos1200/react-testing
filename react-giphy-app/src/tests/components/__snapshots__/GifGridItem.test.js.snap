// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Se debe mostrar correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo del GIF"
    src="https://media.giphy.com/media/CsdYnIX8x44pdUeLBd/giphy-downsized-large.gif"
  />
  <p>
     
    Titulo del GIF
     
  </p>
</div>
`;
